### PR TITLE
Fix/clean last seen environments

### PR DESCRIPTION
**Pull request by FredrikOseberg**
This PR addresses some cleanup related to removing the useLastSeenRefactor flag:

* Added fallback last seen to the feature table last_seen_at column 
* Remove foreign key on environment since we can not guarantee that we will get valid data in this field
* Add environments to cleanup function
* Add test for cleanup environments

### CODE REVIEW

File: src/lib/features/feature-toggle/feature-toggle-strategies-store.ts [hunk 1]
```diff
@@ -386,6 +386,7 @@ class FeatureStrategiesStore implements IFeatureStrategiesStore {
                 acc.description = r.description;
                 acc.project = r.project;
                 acc.stale = r.stale;
+                acc.lastSeenAt = r.last_seen_at;
```
Comment (FredrikOseberg) [inline src/lib/features/feature-toggle/feature-toggle-strategies-store.ts:389]: Fallback to features last_seen_at

File: src/lib/features/feature-toggle/feature-toggle-strategies-store.ts [hunk 2]
```diff
@@ -922,7 +926,8 @@ class FeatureStrategiesStore implements IFeatureStrategiesStore {
             const featureRow = acc[row.feature_name];
             if (
                 featureRow.lastSeenAt === undefined ||
-                new Date(row.env_last_seen_at) > new Date(featureRow.lastSeenAt)
+                new Date(row.env_last_seen_at) >
+                    new Date(featureRow.last_seen_at)
```
Comment (FredrikOseberg) [inline src/lib/features/feature-toggle/feature-toggle-strategies-store.ts:930]: We didn't have this property on the row

File: src/migrations/20231123100052-drop-last-seen-foreign-key.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+exports.up = function (db, cb) {
```
Comment (FredrikOseberg) [inline src/migrations/20231123100052-drop-last-seen-foreign-key.js:1]: Remove foreign key from last_seen_at_metrics

File: src/migrations/20231123100052-drop-last-seen-foreign-key.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  db.runSql(
```
Comment (FredrikOseberg) [inline src/migrations/20231123100052-drop-last-seen-foreign-key.js:2]: Tested this

File: src/migrations/20231123100052-drop-last-seen-foreign-key.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    `
+    ALTER TABLE last_seen_at_metrics DROP CONSTRAINT last_seen_at_metrics_environment_fkey;
+    `,
+    cb
+  );
+};
+
+exports.down = function (db, cb) {
```
Comment (FredrikOseberg) [inline src/migrations/20231123100052-drop-last-seen-foreign-key.js:10]: tested this down migration

**Review by andreas-unleash — APPROVED** [review]
LGTM

File: src/lib/features/feature-toggle/feature-toggle-strategies-store.ts (hunk 2 above)
Comment (sjaanus) [reply src/lib/features/feature-toggle/feature-toggle-strategies-store.ts:930]: Damn, that was my typo. Good catch.

Comment (FredrikOseberg) [reply src/lib/features/feature-toggle/feature-toggle-strategies-store.ts:930]: Yeah, small typo. But the row itself also didn't have last_seen_at. I had to add it to the select columns also for this to work.

**Review by sjaanus — APPROVED** [review]
LG

**Pull request merged**